httpclient5-testing/src/main/java/org/apache/hc/client5/testing/auth/BasicAuthenticationHandler.java

```diff
@@ -27,13 +27,27 @@
 
 package org.apache.hc.client5.testing.auth;
 
+import java.nio.charset.Charset;
 import java.nio.charset.StandardCharsets;
 
 import org.apache.hc.client5.http.auth.StandardAuthScheme;
 import org.apache.hc.client5.http.utils.Base64;
 
 public class BasicAuthenticationHandler extends AbstractAuthenticationHandler {
 
+    private final Charset charset;
+
+    /**
+     * @since 5.3
+     */
+    public BasicAuthenticationHandler(final Charset charset) {
+        this.charset = charset != null ? charset : StandardCharsets.UTF_8;
+    }
+
+    public BasicAuthenticationHandler() {
+        this(StandardCharsets.US_ASCII);
+    }
```

Comment:
Is this for backwards compat? Since it looks inconsistent with no-arg and `null` arg. At least from a client perspective it is suprising.

Reply:
> Is this for backwards compat? 

@michael-o Yes, it is.

Reply:
I see, the for consistency reasons the one-arg constructor should not allow null, encoding must be explicit then. Then the client can explicitly pass `UTF-8` and the rest will always be `US-ASCII`. We could also consider to use always `UTF-8` since ASCII input will always work.